File: N/app/src/main/java/com/example/intent/MainActivity.java
package com.example.intent;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {
    private EditText name,age,email,phone;
    private Button button;

    public final static String
            MESSAGE_KEY ="com.example.message_key";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        name = findViewById(R.id.nameId);
        age = findViewById(R.id.ageId);
        email = findViewById(R.id.emailId);
        phone = findViewById(R.id.phoneId);

        button = findViewById(R.id.buttonId);
        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {

              senData();

            }
        });


    }
    public void senData(){
        String nameTex = name.getText().toString();
        String ageTex = age.getText().toString();
        String emailTex = email.getText().toString();
        String phoneTex = phone.getText().toString();

        Bundle bundle = new Bundle();
        bundle.putString("nameTex", nameTex);
        bundle.putString("ageTex", ageTex);
        bundle.putString("emailTex", emailTex);
        bundle.putString("phoneTex", phoneTex);

        Intent intent = new Intent( this,ShowProfile.class);
        intent.putExtras(bundle);
        startActivity(intent);
        Toast.makeText(this, "Data send perfectly...", Toast.LENGTH_SHORT).show();
    }
}

File: RecyclerViewProject/app/src/main/java/com/example/recyclerviewproject/MainActivity.kt
package com.example.recyclerviewproject

import android.content.Intent
import android.net.Uri
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.example.recyclerviewproject.adapterClass.CollegeListAdapter
import com.example.recyclerviewproject.adapterClass.ItemListener
import com.example.recyclerviewproject.model.CollegeCallBack
import com.example.recyclerviewproject.model.CollegeData
import com.example.recyclerviewproject.model.CollegeModel
import com.example.recyclerviewproject.model.CollegeModelImp

class MainActivity : AppCompatActivity() {
    private lateinit var collegeModel: CollegeModel


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        showCollegeList()
    }

    private fun showCollegeList(){

        collegeModel = CollegeModelImp()
         collegeModel.getCollegeDetails(object : CollegeCallBack {
            override fun onSuccess(collegeList: ArrayList<CollegeData>) {

                val adapter = CollegeListAdapter( collegeList, object : ItemListener {

                    override fun onEmailClicked(position: Int) {
                        val intent = Intent(Intent.ACTION_SENDTO)
                        intent.data = Uri.parse("mailto: shakircam@gmail.com")
                        intent.putExtra(Intent.EXTRA_EMAIL, "test")
                        intent.putExtra(Intent.EXTRA_SUBJECT, "Good Evening")

                        startActivity(intent)
                    }

                    override fun onItemClicked(position: Int) {
                        Toast.makeText(this@MainActivity,"i will do this next day",Toast.LENGTH_SHORT).show()
                    }

                    override fun onNumberClicked(position: Int) {
                        val dialIntent = Intent(Intent.ACTION_DIAL)
                        dialIntent.data = Uri.parse("tel:" + "01917380832")
                        startActivity(dialIntent)
                    }

                })
                val recyclerView = findViewById<RecyclerView>(R.id.recyclerView)

                recyclerView.layoutManager = LinearLayoutManager(this@MainActivity,LinearLayoutManager.VERTICAL,false)
                recyclerView.adapter = adapter
            }

            override fun onError(errorMessage: Throwable) {
                TODO("Not yet implemented")
            }

        })






    }
}